icu4c/source/tools/genrb/reslist.cpp

```diff
@@ -1362,11 +1391,19 @@ SRBRoot::compactKeys(UErrorCode &errorCode) {
     }
     keys = fKeys + fKeysBottom;
     for (; i < keysCount; ++i) {
-        map[i].oldpos = (int32_t)(keys - fKeys);
+        int32_t keyOffset = static_cast<int32_t>(keys - fKeys);
+        while (!fIsPoolBundle && keysInUse.count(keyOffset) == 0) {
```

Comment:
I find the nested loop a little confusing, and unnecessary.

Please change the while to an if, and put the following logic for a remaining key into the else branch. And move the "++i" in there, to count only remaining keys, which also allows changing the for loop to a while loop.

Reply:
Thanks!